src/node_url.cc

```diff
@@ -515,10 +515,10 @@ static inline void PercentDecode(const char* input,
   dest->reserve(len);
   const char* pointer = input;
   const char* end = input + len;
-  size_t remaining = pointer - end - 1;
+
   while (pointer < end) {
     const char ch = pointer[0];
-    remaining = (end - pointer) + 1;
+    size_t remaining = (end - pointer) + 1;
```

Comment:
Oops, it seems the _remaining_ is calculated incorrectly. The [remaining](https://url.spec.whatwg.org/#remaining) must reference the substring after _pointer_, so I guess: 

```c++
size_t remaining = end - (pointer + 1);
```

The current code fails on the following test case:

```c++
std::string dest;
PercentDecode("%21", 1, &dest);
// actual: dest == "!"
// expected: dest == "%" (because len is 1)
```